src/Settings.cpp

```diff
@@ -26,8 +28,15 @@ QVariant Settings::getValue(const std::string& group, const std::string& name)
         return cacheIndex->second;
     } else {
         // Nothing found in the cache, so get the value from the settings file or get the default value if there is no value set yet
-        QSettings settings(QApplication::organizationName(), QApplication::organizationName());
-        QVariant value = settings.value(QString::fromStdString(group + "/" + name), getDefaultValue(group, name));
+        QSettings* settings = NULL;
+
+        if(userConfiguration)
+            settings = new QSettings(userConfigurationDir, QSettings::IniFormat);
```

Comment:
To prevent the known bug, maybe `status()` might help. If the constructor doesn't write the file (which I hope it doesn't), checking status might be useful to avoid writing a file that is not in INI format. Maybe it should abort in that case, so it would be a good ide to check it in the setter.

Reply:
A new file should be allowed, so it might be necessary to check also the existence.

Reply:
I tried to use `status()`, but it didn't help because QSettings constructor write the file first.
How about check first line starts with `[%General]` if the file already exist?

Reply:
Strange design to overwrite the file in the constructor... Maybe using QSettings and setting later the filename and format?

If that won't work, your idea about checking the first line seems sensible.

Reply:
As far as I know that is impossible because as soon as use QSettings the file is created or overwritten.